core/src/main/java/bio/terra/pearl/core/service/publishing/PortalDiffService.java

```diff
@@ -83,15 +88,43 @@ public PortalEnvironmentChange diffPortalEnvs(PortalEnvironment sourceEnv, Porta
             studyEnvChanges.add(studyEnvChange);
         }
 
+        List<ParticipantDashboardAlert> destAlerts = new ArrayList<>(destEnv.getParticipantDashboardAlerts());
+        List<ParticipantDashboardAlert> sourceAlerts = new ArrayList<>(sourceEnv.getParticipantDashboardAlerts());
+        List<ParticipantDashboardAlertChange> alertChangeLists = diffAlertLists(sourceAlerts, destAlerts);
+
         return new PortalEnvironmentChange(
                 siteContentRecord,
                 envConfigChanges,
                 preRegRecord,
                 notificationConfigChanges,
+                alertChangeLists,
                 studyEnvChanges
         );
     }
 
+    protected List<ParticipantDashboardAlertChange> diffAlertLists(
+            List<ParticipantDashboardAlert> sourceAlerts,
+            List<ParticipantDashboardAlert> destAlerts) throws Exception {
```

Comment:
What exception does this throw and where is it handled?

Reply:
Just wanted to followup on this since I realize from another PR that my comments about exceptions may not be clear. That seems likely here, so my apologies for not being more specific.

This is my POV, so feel free to share your thoughts with me if they differ, but throwing `Exception` feels odd here. Library methods will often list the exceptions they throw since it is a way to communicate specific error conditions to callers, and if that's your aim here then I would suggest listing the specific exceptions you want callers to be aware of. You also must list any exceptions that do not inherit from RuntimeException -- that is, checked exceptions -- but again it is best to list the specific exceptions.
Many of our methods throw exceptions and do not list them,  since they do not expect any specific handling and are happy with the way those exceptions are handled in the GlobalExceptionHandler. If that's the case here, you can just omit the throws clause.

Does that make sense?

Reply:
I appreciate the clarification, that does help me understand more. A lot of the methods in this class and related classes throw a broad `Exception`, so that's why I'm having to throw `Exception` here. My intent was to just let it bubble up and let the GlobalExceptionHandler deal with it -- there's not any special handling that I want to do here.

This part of the codebase should probably be combed over to make the exception handling more clear. In the meantime, I've updated the exceptions in this particular code path to (I hope) be a bit more clear (see 32923897f1988c77479b892fac9e02b83fb6f05e)